tests/test_ote_api.py

```diff
@@ -285,16 +285,24 @@ def test_cancel_training_detection(self):
             model_status=ModelStatus.NOT_READY
         )
 
+        training_progress_curve = []
+        def progress_callback(progress: float, score: Optional[float] = None):
+            training_progress_curve.append(progress)
+
+        train_parameters = TrainParameters
+        train_parameters.update_progress = progress_callback
+
         # Test stopping after some time
         start_time = time.time()
-        train_future = executor.submit(detection_task.train, dataset, output_model)
+        train_future = executor.submit(detection_task.train, dataset, output_model, train_parameters)
         # give train_thread some time to initialize the model
         while not detection_task._is_training:
             time.sleep(10)
         detection_task.cancel_training()
 
         # stopping process has to happen in less than 35 seconds
         train_future.result()
+        assert training_progress_curve[-1] == 100
```

Comment:
```suggestion
        self.assertEqual(training_progress_curve[-1], 100)
```

Reply:
>       assert training_progress_curve[-1] == 101
E       AssertionError: assert 100.0 == 101
E         +100.0
E         -101

>       self.assertEqual(training_progress_curve[-1], 101)
E       AssertionError: 100.0 != 101

What do you like more?

Reply:
The latter one.